includes/Integrations/Core_Themes_Support.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Class Core_Themes_Support
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Integrations;
+
+use Google\Web_Stories\Customizer;
+
+/**
+ * Class Core_Themes_Support.
+ */
+class Core_Themes_Support {
+
+	/**
+	 * Current theme
+	 *
+	 * @var string
+	 */
+	protected $template = '';
+
+	/**
+	 * Default array of core themes to add support to.
+	 *
+	 * @var array
+	 */
+	protected static $supported_themes = [
+		'twentytwentyone',
+		'twentytwenty',
+		'twentynineteen',
+		'twentyseventeen',
+		'twentysixteen',
+		'twentyfifteen',
+		'twentyfourteen',
+		'twentythirteen',
+		'twentytwelve',
+		'twentyeleven',
+		'twentyten',
+		'classic',
+		'default',
+	];
+
+	/**
+	 * Get list of supported core themes.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return string[] Slugs for supported themes.
+	 */
+	public static function get_supported_themes() {
+		return self::$supported_themes;
+	}
+
+	/**
+	 * Adds theme support for Web Stories.
+	 *
+	 * This will enable add_theme_support with predefined
+	 * options supported themes.
+	 *
+	 * @since 1.3.0
+	 */
+	public function extend_theme_support() {
+		add_theme_support( 'web-stories' );
+	}
+
+	/**
+	 * Embed Webstories.
+	 *
+	 * Embeds web stories with default customizer settings.
+	 *
+	 * @since 1.3.0
+	 */
+	public function embed_web_stories() {
+
```

Comment:
Core themes Check is missing here.

Use latest commit - https://github.com/rtCamp/web-stories-wp/compare/rt-dev...rtCamp:feature/add-core-themes-customizer-support#diff-87b8eed936695c54c05fadf648eae322df4799ec5def11d565070e12d97d9426R89-R92

Reply:
Core theme check is added in the `init()` method as we wouldn't want to run this class if not core theme.

https://github.com/rtCamp/web-stories-wp/blob/62dd53716ae4dbf852ff6eeb74dac7e0eb1073e9/includes/Integrations/Core_Themes_Support.php#L125